diff --git a/app/config/services.yml b/app/config/services.yml
index 62acbf2..71f8aa1 100644
--- a/app/config/services.yml
+++ b/app/config/services.yml
@@ -1,5 +1,6 @@
 services:
     app_cookie_voter:
         class: AppBundle\Security\CookieVoter
+        arguments: ["@service_container"]
         tags:
             - { name: security.voter }
diff --git a/src/AppBundle/Security/CookieVoter.php b/src/AppBundle/Security/CookieVoter.php
index 574aca4..f25581a 100644
--- a/src/AppBundle/Security/CookieVoter.php
+++ b/src/AppBundle/Security/CookieVoter.php
@@ -2,11 +2,19 @@
 
 namespace AppBundle\Security;
 
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\AbstractVoter;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 class CookieVoter extends AbstractVoter
 {
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
     /**
      * Return an array of supported classes. This will be called by supportsClass
      *
@@ -46,6 +54,12 @@ class CookieVoter extends AbstractVoter
             return false;
         }
 
+        $authChecker = $this->container->get('security.authorization_checker');
+
+        if ($authChecker->isGranted('ROLE_COOKIE_MONSTER')) {
+            return true;
+        }
+
         if ($object->getBakerUsername() == $user->getUsername()) {
             return true;
         }
